src/psp.ts

```diff
@@ -0,0 +1,502 @@
+            continue;
+          }
+
+          // **************************************************************************
+          // Make API call
+          // **************************************************************************
+          node.logger.debug('Making API requests...', subscriptionIdLogOptions);
+
+          const [error, logsData] = await retryGo(() =>
+            readApiValue({
+              airnodeAddress,
+              oises,
+              apiCredentials,
+              id: subscriptionId,
+              trigger: subscription,
+            })
+          );
+          if (!isNil(error) || isNil(logsData)) {
+            node.logger.warn('Failed to fecth API value', subscriptionIdLogOptions);
+            continue;
+          }
+          const [logs, data] = logsData;
+          node.logger.logPending(logs, subscriptionIdLogOptions);
+
+          const apiValue = data[subscriptionId];
+          if (isNil(apiValue)) {
+            node.logger.warn('API value is missing. Skipping update...', subscriptionIdLogOptions);
+            continue;
+          }
+
+          // **************************************************************************
+          // Check conditions
+          // **************************************************************************
+          node.logger.debug('Checking conditions...', subscriptionIdLogOptions);
+
+          // Is this step really needed?
+          // https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-protocol-v1/contracts/dapis/DapiServer.sol#L20-L23
+
+          const encodedFulfillmentData = ethers.utils.defaultAbiCoder.encode(['int256'], [apiValue]);
+
+          // TODO: Should "subscription.conditions" be already encoded in config?
+          // const encodedConditionParameters = ethers.utils.defaultAbiCoder.encode(
+          //   ['uint256'],
+          //   [subscription.conditions] // How is this supposed to be set in config? i.e. (10**18 / 10) for 10%?
+          // );
+
+          const voidSigner = new ethers.VoidSigner(ethers.constants.AddressZero, provider);
+          // TODO: retryGo?
+          if (
+            !(await dapiServer
+              .connect(voidSigner)
+              .conditionPspBeaconUpdate(subscriptionId, encodedFulfillmentData, subscription.conditions))
+          ) {
+            node.logger.warn('Conditions not met. Skipping update...', subscriptionIdLogOptions);
+            continue;
+          }
+          node.logger.info('Conditions met. Updating beacon...', subscriptionIdLogOptions);
+
+          // **************************************************************************
+          // Compute signature
+          // **************************************************************************
+          node.logger.debug('Signing fulfill message...', subscriptionIdLogOptions);
+
+          const airnodeWallet = ethers.Wallet.fromMnemonic(nodeSettings.airnodeWalletMnemonic).connect(provider);
```

Comment:
Do this once globally because this is probably computationally intensive. You shouldn't need to connect the provider to sign a message so this is chain/provider agnostic.